lntest/harness_assertion.go

```diff
@@ -1639,6 +1639,19 @@ func (h *HarnessTest) AssertPaymentStatus(hn *node.HarnessNode,
 	return target
 }
 
+// AssertPaymentFailureReason asserts that the given node lists a payment with
+// the given preimage which has the expected failure reason.
+func (h *HarnessTest) AssertPaymentFailureReason(hn *node.HarnessNode,
```

Comment:
think this assertion is unnecessary since we always have the payment and can assert the fields on it directly.

Reply:
not addressed

Reply:
I've removed the `Predicate` but for mentioned reason I couldn't remove the assertion.